zipline/utils/calendars/trading_calendar.py

```diff
@@ -176,7 +256,7 @@ def next_open(self, dt):
             The UTC timestamp of the next open.
         """
         idx = next_divider_idx(self.market_opens_nanos, dt.value)
-        return self.schedule.market_open[idx].tz_localize('UTC')
+        return pd.Timestamp(self.market_opens_nanos[idx], tz='UTC')
```

Comment:
sorry, I should have commented, or put it in a separate commit - this is a performance improvement.  The new line runs ~8x faster than the old line.